### DISCUSSION TITLE
Works well with EeasyMeter Q3DA3002 V3.04

### DISCUSSION THREAD

**Discussion by martin8310** [mampfes/esphome_obis_d0#19] [General]
Hello mampfes,

that works also great with my smartmeter ;-) Thanks.

below my ESPHome Code:

#-----------------------------CODE EasyMeter Q3D, Q3DA1024 (OBIS (d0 Protokoll))~ 9600 baud (9600 7E1). OBIS ASCII type of coding ----------------------------------

uart:
  id: uart_bus
  rx_pin: GPIO16
  baud_rate: 9600
  data_bits: 7
  parity: EVEN
  stop_bits: 1
  #debug:

external_components:
  - source: github://mampfes/esphome_obis_d0

obis_d0:
  id: my_sm
  uart_id: uart_bus
  on_telegram:
    then:
      - logger.log: "telegram received"

sensor:
  - platform: wifi_signal # Reports the WiFi signal strength/RSSI in dB
    name: "WiFi Signal dB"
    id: wifi_signal_db
    update_interval: 60s
    entity_category: "diagnostic"

  - platform: copy # Reports the WiFi signal strength in %
    source_id: wifi_signal_db
    name: "WiFi Signal Percent"
    filters:
      - lambda: return min(max(2 * (x + 100.0), 0.0), 100.0);
    unit_of_measurement: "Signal %"
    entity_category: "diagnostic"
    device_class: ""

  - platform: obis_d0
    name: "Zaehlerstand"
    obis_d0_id: my_sm
    obis_code: "1-0:1.8.0*255"
    unit_of_measurement: kWh
    accuracy_decimals: 3
    state_class: total_increasing
    device_class: energy
    value_regex: "\\d{8}\\.\\d{7}\\*kWh"

  - platform: obis_d0
    name: "Gesamtleistung"
    obis_d0_id: my_sm
    obis_code: "1-0:1.7.0*255"
    unit_of_measurement: W
    accuracy_decimals: 0
    device_class: power
    #device_class: voltage
    #device_class: current
    #device_class: frequency
    state_class: measurement
    
  - platform: obis_d0
    name: "Leistung L1"
    obis_d0_id: my_sm
    obis_code: "1-0:21.7.0*255"
    unit_of_measurement: W
    accuracy_decimals: 0
    device_class: power
    #device_class: voltage
    #device_class: current
    #device_class: frequency
    state_class: measurement

  - platform: obis_d0
    name: "Leistung L2"
    obis_d0_id: my_sm
    obis_code: "1-0:41.7.0*255"
    unit_of_measurement: W
    accuracy_decimals: 0
    device_class: power
    #device_class: voltage
    #device_class: current
    #device_class: frequency
    state_class: measurement

  - platform: obis_d0
    name: "Leistung L3"
    obis_d0_id: my_sm
    obis_code: "1-0:61.7.0*255"
    unit_of_measurement: W
    accuracy_decimals: 0
    device_class: power
    #device_class: voltage
    #device_class: current
    #device_class: frequency
    state_class: measurement

text_sensor:
  - platform: obis_d0
    name: "Device Identification"
    obis_d0_id: my_sm
    obis_code: "0-0:96.1.255*255"
    entity_category: diagnostic
    value_regex: "\\w{14}"

  - platform: obis_d0
    name: "Manufacturer ID"
    obis_d0_id: my_sm
    obis_code: "id"
    entity_category: diagnostic
    #value_regex: "\\w+"

**Comment by mampfes** [comment]
Thanks for sharing. I will add your example to the README.
